content/blog/deploy-stateful-mongodb-application-on-kubernetes-clusters-in-hpe-greenlake-for-containers-1.md

```diff
@@ -0,0 +1,547 @@
+
+```markdown
+apiVersion: v1
+kind: Service
+metadata:
+  name: mongodb
+  labels:
+    app: mongo
+spec:
+  ports:
+  - port: 27017
+    protocol: TCP
+    targetPort: 27017
+  selector:
+    app: mongo
+  clusterIP: None
+```
+
+### Deploy MongoDB Application with Kustomize
+
+With all created YAML files, MongoDB application will be deployed using [Kustomize](https://kustomize.io/).
+
+#### 1. Move all YAML files to the folder '*base*':
+
+```markdown
+├── base
+│   ├── clusterrolebinding.yaml
+│   ├── clusterrole.yaml
+│   ├── kustomization.yaml
+│   ├── namespace.yaml
+│   ├── pvc.yaml
+│   ├── sa.yaml
+│   ├── service.yaml
+│   └── statefulset.yaml
+```
+
+The file *kustomization.yaml* lists all YAML files in its *resources* section:
+
+```markdown
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - namespace.yaml
+  - sa.yaml
+  - clusterrole.yaml
+  - clusterrolebinding.yaml
+  - statefulset.yaml
+  - service.yaml
+```
+
+#### 2. Create another folder '*overlays*' and a sub-folder '*production*':
+
+```markdown
+└── overlays
+    └── production
+        ├── kustomization.yaml
+        └── patch-statefulset.yaml
+```
+
+The contents of the file *kustomization.yaml*:
+
+```markdown
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+bases:
+- ../../base
+commonLabels:
+  env: production
+namePrefix: pce-
+namespace: pce-mongodb
+patches:
+- ./patch-statefulset.yaml
+```
+
+The contents of the file *patch-statefulset.yaml*:
+
+```markdown
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: mongodb-replica
+spec:
+  replicas: 3
+```
+
+This YAML file will patch the original StatefulSet and deploy the MongoDB application as a three-member replica set.
+
+#### 3. Deploy MongoDB with the following command:
+
+```markdown
+$ kubectl apply -k overlays/production
+namespace/pce-mongodb created
+serviceaccount/pce-mongo-account created
+clusterrole.rbac.authorization.k8s.io/pce-mongo-role created
+clusterrolebinding.rbac.authorization.k8s.io/pce-mongo_role_binding created
+service/pce-mongodb created
+statefulset.apps/pce-mongodb-replica created
+```
+
+MongoDB application is deployed as Replica Set to the namespace '*pce-mongodb*' using the service account '*pce-mongo-account*'. Running the following command, you should see three specified replicas pods are all in **running** status: 
+
+```markdown
+$ kubectl get all -n pce-mongodb
+NAME                        READY   STATUS    RESTARTS   AGE
+pod/pce-mongodb-replica-0   1/1     Running   0          30s
+pod/pce-mongodb-replica-1   1/1     Running   0          19s
+pod/pce-mongodb-replica-2   1/1     Running   0          11s
+
+NAME                  TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)     AGE
+service/pce-mongodb   ClusterIP   None         <none>        27017/TCP   30s
+
+NAME                                   READY   AGE
+statefulset.apps/pce-mongodb-replica   3/3     30s
+```
+
+## Set up MongoDB Replication Host
```

Comment:
Is MongoDB Replication Host a proper noun? Or should it be MongoDB replication host?